pyface/workbench/workbench_window.py

```diff
@@ -849,16 +849,15 @@ def _views_items_changed(self, event):
 
     # Dynamic ----
 
-    @on_trait_change("layout.editor_closed")
-    def _on_editor_closed(self, editor):
+    @observe("layout:editor_closed")
```

Comment:
I believe the `.` from before was a bug and should have been a `:`.  It seems from the handler itself that it was assumed editor was in fact an `IEditor` instance as we ultimately call `self.editors.index(editor)`.  Without this change I was not seeming test failuires, but exceptions were being raised in `test_workbench_window.py` saying a `MagicMock` instance was not in `self.editors`.

This made me learn a new behavior of otc.  Previously, when we set `workbench_window.layout = mock.MagicMock(spec=WorkbenchWindowLayout)` this handler was invoked.  However, by modifying the signature I saw that `name` was actually `editor_closed` not `layout` even though layout was changing.  This caused the `editor` argument to come up as `Undefined` when running the test thus avoiding the `self.editors.index(editor)` line.  When we make the switch, `event.name` is now `layout` not `editor_changed` and so `event.new != Undefined` as one might have suspected based on what happened before.

I'm not sure if this is explicitly documented in the old otc docs?  Or if this is a bug?  In any case I believe the new code is correct